glass/user.py

```diff
@@ -55,23 +74,32 @@ class _FitsWriter:
     Initialised with the fits object and extension name.
     """
 
-    def __init__(self, fits, ext=None) -> None:  # type: ignore[no-untyped-def]
+    def __init__(self, fits: fitsio.FITS, ext: str = "") -> None:
         """Create a new, uninitialised writer."""
         self.fits = fits
         self.ext = ext
 
-    def _append(self, data, names=None) -> None:  # type: ignore[no-untyped-def]
+    def _append(
+        self,
+        data: npt.NDArray[np.float64] | list[npt.NDArray[np.float64]],
+        names: list[str] | None = None,
+    ) -> None:
         """Write the FITS file."""
-        if self.ext is None or self.ext not in self.fits:
+        if not self.ext or self.ext not in self.fits:
```

Comment:
These are not the same, `ext` can be ""

Reply:
https://github.com/glass-dev/glass/pull/368#discussion_r1831121217